pkg/topology/kademlia/kademlia.go

```diff
@@ -454,7 +455,11 @@ func (k *Kad) manage() {
 	go func() {
 		<-k.quit
 		if !timer.Stop() {
-			<-timer.C
+			select {
+			case <-timer.C:
+			case <-time.After(1 * time.Second):
+				k.logger.Debugf("kademlia timer not drained after 1 second, goroutine dump...\n%s\n*** end\n", goroutine.Dump())
```

Comment:
why not use `pprof.Lookup("goroutine").WriteTo(os.Stdout, 1)`?

Reply:
fixed it to use pprof